pandas/tests/generic/methods/test_sample.py

```diff
@@ -66,69 +66,78 @@ def test_sample_lengths(self, obj):
 
     def test_sample_invalid_random_state(self, obj):
         # Check for error when random_state argument invalid.
-        with pytest.raises(ValueError):
-            obj.sample(random_state="astring!")
+        msg = (
+            "random_state must be an integer, array-like, a BitGenerator, a numpy "
+            "RandomState, or None"
+        )
+        with pytest.raises(ValueError, match=msg):
+            obj.sample(random_state="a_string")
```

Comment:
any reason why you removed the exclamation mark `!`?

Reply:
I'm not saying it needs to be changed back, it's fine like this, was just wondering if it had any effect

Reply:
I guess it's not really a good reason - but I didn't want people to think that it was including punctuation that made it fail. I wanted to make it clear that what was failing was that it was any string at all.

In a couple of other places in various related PRs I changed random things like "aasdf" to "invalid_plot_kind" or whatever just to make the test clearer and this goes along with that.

Reply:
sure, makes sense - thanks for all your help with these tests!